addons/stock_account/models/product.py

```diff
@@ -202,7 +202,7 @@ def _change_standard_price(self, new_price, counterpart_account_id=False):
             if product.cost_method not in ('standard', 'average'):
                 continue
             quantity_svl = product.sudo().quantity_svl
-            if float_is_zero(quantity_svl, precision_rounding=product.uom_id.rounding):
+            if float_is_zero(quantity_svl, precision_rounding=product.uom_id.rounding) or quantity_svl < 0:
```

Comment:
You can use `float_compare(...) <= 0` to be sure the rounding is done for negative quantities too

Reply:
It's been updated, thanks :)